### ISSUE TITLE
LFilter and LFilterPyTorch provide inconsistent gradients

### ISSUE THREAD

**Issue by davidslater** [Trusted-AI/adversarial-robustness-toolbox#992]
**Describe the bug**
LFilter and LFilterPyTorch provide inconsistent gradients. In particular, the LFilter ones are incorrect.

**To Reproduce**
```
import numpy as np
import torch
from torch.autograd import Variable
from art.preprocessing.audio import LFilter, LFilterPyTorch

f = LFilter(numerator_coef=np.array([1.0, 0.0, 1.0]), denominator_coef=np.array([1.0, 0.0, 0.0]))
f_py = LFilterPyTorch(numerator_coef=np.array([1.0, 0.0, 1.0]), denominator_coef=np.array([1.0, 0.0, 0.0]))
x = Variable(torch.randn(10), requires_grad=True)

f.estimate_gradient(x.detach().numpy(), grad=np.ones(10))
# array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])

f_py.estimate_gradient(x.detach().numpy(), grad=np.ones(10))
# array([2., 2., 2., 2., 2., 2., 2., 2., 1., 1.], dtype=float32)

# This estimate is equivalent to:
z, _ = f_py.forward(x)
z.backward(torch.ones(10))
x.grad
```

**Expected behavior**
LFilter is using identity BPDA to estimate the gradient, but this is incorrect. I'm not sure how much work it is to implement for IIR filters, but FIR should be pretty straightforward.

**System information (please complete the following information):**
ART version 1.6.0

**Comment by beat-buesser** [comment]
Hi @davidslater Thank you very much for raising this issue!

@minhitbk What do you think?
